src/UI/Component/Chart/Speedo/Speedo.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/* Copyright (c) 2017 Ralph Dittrich <dittrich@qualitus.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Chart\Speedo;
+use ILIAS\UI\Component\Component;
+
+/**
+ * Interface Speedo
+ * @package ILIAS\UI\Component\Chart\Speedo
+ */
+interface Speedo extends Component {
+
+    /**
+     * Get maximum value
+     *
+     * This value is used as 100%.
+     * This value will always returned "raw" because it is used to calculate the
+     * percentage values of score, minimum and diagnostic.
+     *
+     * @return int|float
+     */
+    public function getMaximum();
+
+    /**
+     * Get minimum value
+     *
+     * This value represents the minimum score that users need, to fulfill the objective.
+     * If this value is not set, it defaults to 100%.
+     *
+     * @param bool $getAsPercent Get Value as percentage value or not
+     * @return int|float
+     */
+    public function getMinimum($getAsPercent = true);
+
+    /**
+     * Get user score value
+     *
+     * This value represents the users score. It is rendered as the speedo main bar.
+     *
+     * @param bool $getAsPercent Get Value as percentage value or not
+     * @return int|float
+     */
+    public function getScore($getAsPercent = true);
+
+    /**
+     * Test if a diagnostic score is given
+     *
+     * This should always be used before you try to get the value.
+     *
+     * @return bool
+     */
+    public function hasDiagnostic();

```

Comment:
I know, I'm late to answer here but to clearify this and prevent the same question in the future: The diagnostic score (in the future "comparision value") has nothing to do with the "Mini" version of this component. It decides wether the component shows one or two bars in the Standard/Responsive (in the future "Standard/FixedSize") version. If a comparision value is given, the component shows two bars, otherwise one.
I have written this function to the interface because if someone will extend or modify the component in the future, this distinction may be very important to correctly render the given values.
If you do not share my opinion, we can surely discuss it again in the new PR. I will link it asap in a comment on this PR, so it will be easy to find.